presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PlanNodeDecorrelator.java

```diff
@@ -189,6 +218,78 @@ public PlanNodeDecorrelator(PlanNodeIdAllocator idAllocator, Lookup lookup)
                     true));
         }
 
+        private Optional<DecorrelationResult> rewriteLimitWithRowCountGreaterThanOne(DecorrelationResult childDecorrelationResult, long limit, PlanNodeId nodeId)
+        {
+            PlanNode decorrelatedChildNode = childDecorrelationResult.node;
+            Set<Symbol> constantSymbols = childDecorrelationResult.getConstantSymbols();
+
+            if (constantSymbols.isEmpty() || !constantSymbols.containsAll(childDecorrelationResult.symbolsToPropagate)) {
```

Comment:
AFAIR this condition was required in `limit -> aggregation` rewrite to avoid creating default aggregations. Maybe that conditions should be refactored?